client/src/components/SignupForm.jsx

```diff
@@ -44,24 +44,19 @@ function SignupForm({ setSignupSuccess }) {
     async function handleSubmit(e) {
         e.preventDefault();
 
-        try {
-            const res = await signupUser(userData);
+        const res = await signupUser(userData);
 
-            if (
-                res.success === 'false' &&
-                res.errorType === 'VALIDATION_ERROR'
-            ) {
-                setErrors(res.errors);
-                return;
-            } else if (res.success === 'false') {
-                setErrors({ generic: 'something went wrong' });
-                return;
-            }
-            setSignupSuccess(true);
-        } catch (error) {
-            console.log(error);
-            setErrors({ generic: 'something went wrong' });
+        if (res.success === 'false' && res.errorType === 'VALIDATION_ERROR') {
```

Comment:
```suggestion
        if (res.success === false && res.errorType === 'VALIDATION_ERROR') {
```

change this to boolean instead of string